framework/src/kernels/NonlocalKernel.C

```diff
@@ -39,6 +39,58 @@ NonlocalKernel::NonlocalKernel(const InputParameters & parameters) :
 }
 
 void
+NonlocalKernel::computeJacobian()
+{
```

Comment:
I gave it some thought actually. My initial intention was to keep these in Kernel itself. However, for optimization purpose I had to alter few things and I did not want to disturb the Kernel class.